BrainPortal/app/controllers/groups_controller.rb

```diff
@@ -194,20 +213,23 @@ def unregister
         flash[:notice] = "You have been unregistered from project #{@group.name}."
         format.html { redirect_to :action => "index" }
         format.xml  { head :ok }
+        format.json { head :ok}
       end
     end
   end
 
   # DELETE /groups/1
   # DELETE /groups/1.xml
+  # DELETE /groups/1.json
   def destroy  #:nodoc:
-    @group = current_user.available_groups.find(params[:id])
+    @group = current_user.available_groups.where(params[:id]).first
```

Comment:
Leave the `find`, which will raise an exception if the geroup isn't found and will trigger the appropriate code in the application controller!